src/kanata/linux.rs

```diff
@@ -192,3 +198,60 @@ fn handle_scroll(
         _ => unreachable!("expect to be handling a wheel event"),
     }
 }
+
+fn start_event_preprocessor(
+    preprocess_rx: Receiver<KeyEvent>,
+    process_tx: Sender<KeyEvent>,
+    debounce_duration: Duration,
+) {
+    let mut last_key_event_time: HashMap<OsCode, Instant> = HashMap::new();
+
+    std::thread::spawn(move || {
+        loop {
+            match preprocess_rx.try_recv() {
+                Ok(kev) => {
+                    let now = Instant::now();
+                    let oscode = kev.code;
+
+                    match kev.value {
+                        KeyValue::Release => {
+                            // Always allow key releases to pass through
```

Comment:
This debouncing algorithm will mean that if a "bounce" happens, a key will be tapped and be impossible to be held. The code should reference and implement an established debouncing algorithm.